src/App.tsx

```diff
@@ -1,24 +1,131 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { FilterTodos } from './components/FilterTodos';
+import { Form } from './components/Form';
+import { NotificationError } from './components/NotificationError';
 
-const USER_ID = 0;
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+
+const USER_ID = 10210;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setError(ErrorType.LOAD);
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+  };
+
+  const postTodo = useCallback(async (title: string) => {
+    try {
+      const newTodo = {
+        userId: USER_ID, completed: false, title,
+      };
+      const temp = { ...newTodo, id: 0 };
+
+      setLoading(true);
+      setTempTodo(temp);
+      await addTodo(newTodo);
+      await getTodosFromServer();
+    } catch {
+      setError(ErrorType.ADD);
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    } finally {
+      setTempTodo(null);
+      setLoading(false);
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (todoId: number) => {
+    try {
+      await deleteTodo(todoId);
+      await getTodosFromServer();
+    } catch {
+      setError(ErrorType.DELETE);
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.ACTIVE:
+          return !todo.completed;
+
+        case FilterType.COMPLETED:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+  }, [todos, filterType]);
```

Comment:
No fixed. No need to filter in case FilterType is All

Reply:
You mean something like this?

  ```
const filteredTodos = useMemo(() => {
    if (filterType === FilterType.ALL) {
      return todos;
    }


    return todos.filter(todo => {
      switch (filterType) {
        case FilterType.ACTIVE:
          return !todo.completed;

        case FilterType.COMPLETED:
          return todo.completed;

        default:
          return true;
      }
    });
}, [todos, filterType]);
```

or this?

```
  const filteredTodos = useMemo(() => {
    switch (filterType) {
      case FilterType.ACTIVE:
        return todos.filter(todo => !todo.completed);
      case FilterType.COMPLETED:
        return todos.filter(todo => todo.completed);
      default:
        return todos;
    }
  }, [todos, filterType]);